### DISCUSSION TITLE
STM32H5xx SPI DMA

### DISCUSSION THREAD

**Discussion by NagaHimanshu** [zephyrproject-rtos/zephyr#71871] [Q&A] [answered]
Hi, I am working on SPI DMA transfers on STM32H563ZI. I was able to transfer data over SPI using DMA and I observed that it is a blocking transfer and core waits till transfer is complete. The polling sequence are as follows:

https://github.com/zephyrproject-rtos/zephyr/blob/ca4ff1e6fe97dc00e316d6833ac4ee5c01da43e2/drivers/spi/spi_ll_stm32.c#L1061

https://github.com/zephyrproject-rtos/zephyr/blob/ca4ff1e6fe97dc00e316d6833ac4ee5c01da43e2/drivers/spi/spi_ll_stm32.h#L223

Is there any support for non-blocking SPI DMA transfer?

**Comment by ajarmouni-st** [comment, marked as answer]
Hi @NagaHimanshu, for now non-blocking DMA transfers are not supported, but I'm working on it, hopefully will submit a PR next week.

**Reply by NagaHimanshu** [reply]
Hi @ajarmouni-st, thanks for letting me know.

**Reply by NagaHimanshu** [reply]
Hi @ajarmouni-st, any update on this?

**Reply by ajarmouni-st** [reply]
Hi @NagaHimanshu, yes I have been working on it since last week, here is what I have for now https://github.com/ajarmouni-st/zephyr_dev/tree/dev_stm32_spi_async, not yet ready to be submitted since the second buffer in `async call` test is not received correctly, only half of it (18 bytes) is. Tested on nucleo_h753zi, nucelo_f767zi & nucleo_f429zi.
To reproduce the result `west build -p always -b nucleo_h753zi tests/drivers/spi/spi_loopback -T drivers.spi.stm32_spi_async_dma.loopback`

**Reply by NagaHimanshu** [reply]
@ajarmouni-st Thanks for the reply. I will try to reproduce the result from the branch. I hope it will be resolved at the earliest.

**Reply by ajarmouni-st** [reply]
Hi @NagaHimanshu, the loopback test now passes with no errors on `nucleo_f767zi` & `nucleo_f429zi`.

**Reply by NagaHimanshu** [reply]
Thanks for the update @ajarmouni-st.
